harness/determined/exec/prep_container.py

```diff
@@ -0,0 +1,66 @@
+import argparse
+import base64
+import io
+import os
+import tarfile
+
+import determined as det
+from determined import constants
+from determined.common.api import certs, request
+
+
+def trial_prep(info: det.ClusterInfo, cert: certs.Cert) -> None:
+    trial_info = det.TrialInfo._from_env()
+    trial_info._to_file()
+
+    resp = request.get(info.master_url, f"api/v1/experiments/{trial_info.experiment_id}/model_def")
```

Comment:
because the server might provide a cert.  This is the case with _all_ tls masters, though technically its only necessary in self-signed certificate situations.